pkg/controller/dns/consts_for_test.go

```diff
@@ -22,6 +22,21 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+type dnsTestCase struct {
+	name                       string
+	existingResources          []client.Object
+	crd                        func() ExternalDNSCRDConfiguration
+	expectedUserError          string
+	expectedError              error
+	expectedDeployment         func() *appsv1.Deployment
```

Comment:
why are these fns?

Reply:
You can call the function or make helper functions to do that.

https://github.com/Azure/aks-app-routing-operator/blob/d62fa0178e2086b9115a28bf6041967e6b83225b/pkg/manifests/nginx_test.go#L341